### PR TITLE
deadbeef/plugins: add the waveform seekbar plugin 

### LINKED ISSUE #52612: Deadbeef: add the waveform seekbar plugin
This one: https://github.com/cboxdoerfer/ddb_waveform_seekbar

### PR DESCRIPTION
**Pull request by suhr**
Fixes https://github.com/NixOS/nixpkgs/issues/52612.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I don't know how to do `nix-build` of a deadbeef plugin.

### CODE REVIEW

**Comment by timokau** [comment]
@GrahamcOfBorg eval

**Comment by suhr** [comment]
> the checksum is required

But how do I know the valid checksum?

**Comment by Mic92** [comment]
@suhr we use the trust-on-first-use model for most sources by specifying an incorrect checksum on the first use or using something like `nix-prefetch-url`.

File: pkgs/applications/audio/deadbeef/plugins/waveform.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    cp gtk3/ddb_misc_waveform_GTK3.so $out/lib/deadbeef
+
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Waveform Seekbar plugin for DeaDBeeF audio player";
+    homepage = https://github.com/cboxdoerfer/ddb_waveform_seekbar;
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.jtojnar ];
```
Comment (jtojnar) [inline pkgs/applications/audio/deadbeef/plugins/waveform.nix]: Would you be willing to maintain this?

Comment (suhr) [reply pkgs/applications/audio/deadbeef/plugins/waveform.nix]: Probably not. But it doesn't need an active maintaining, I believe.

**Pull request closed without merging**